ringhold/src/intTest/java/org/opencadc/ringhold/InventoryValidatorTest.java

```diff
@@ -0,0 +1,328 @@
+ *  License as published by the          License” telle que publiée
+ *  Free Software Foundation,            par la Free Software Foundation
+ *  either version 3 of the              : soit la version 3 de cette
+ *  License, or (at your option)         licence, soit (à votre gré)
+ *  any later version.                   toute version ultérieure.
+ *
+ *  OpenCADC is distributed in the       OpenCADC est distribué
+ *  hope that it will be useful,         dans l’espoir qu’il vous
+ *  but WITHOUT ANY WARRANTY;            sera utile, mais SANS AUCUNE
+ *  without even the implied             GARANTIE : sans même la garantie
+ *  warranty of MERCHANTABILITY          implicite de COMMERCIALISABILITÉ
+ *  or FITNESS FOR A PARTICULAR          ni d’ADÉQUATION À UN OBJECTIF
+ *  PURPOSE.  See the GNU Affero         PARTICULIER. Consultez la Licence
+ *  General Public License for           Générale Publique GNU Affero
+ *  more details.                        pour plus de détails.
+ *
+ *  You should have received             Vous devriez avoir reçu une
+ *  a copy of the GNU Affero             copie de la Licence Générale
+ *  General Public License along         Publique GNU Affero avec
+ *  with OpenCADC.  If not, see          OpenCADC ; si ce n’est
+ *  <http://www.gnu.org/licenses/>.      pas le cas, consultez :
+ *                                       <http://www.gnu.org/licenses/>.
+ *
+ *  : 5 $
+ *
+ ************************************************************************
+ */
+
+package org.opencadc.ringhold;
+
+import ca.nrc.cadc.db.ConnectionConfig;
+import ca.nrc.cadc.db.DBConfig;
+import ca.nrc.cadc.db.DBUtil;
+import ca.nrc.cadc.util.FileUtil;
+import ca.nrc.cadc.util.HexUtil;
+import ca.nrc.cadc.util.Log4jInit;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.net.URI;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Date;
+import java.util.Map;
+import java.util.MissingResourceException;
+import java.util.Properties;
+import java.util.TreeMap;
+import java.util.UUID;
+import javax.sql.DataSource;
+import org.apache.log4j.Level;
+import org.apache.log4j.Logger;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.opencadc.inventory.Artifact;
+import org.opencadc.inventory.DeletedStorageLocationEvent;
+import org.opencadc.inventory.db.ArtifactDAO;
+import org.opencadc.inventory.db.DeletedStorageLocationEventDAO;
+import org.opencadc.inventory.db.SQLGenerator;
+import org.opencadc.inventory.db.version.InitDatabase;
+import org.springframework.jdbc.core.JdbcTemplate;
+
+/**
+ * Various versions of:
+ * Insert artifacts more than uri pattern
+ * Run tool with one uri deselector
+ * Confirm delete storage location event creation and absence of artifacts in inventory
+ */
+public class InventoryValidatorTest {
```

Comment:
There should be a test with a faulty (dangerous) config that verifies that this will not remove a bunch of content when misconfigured. I'm not concerned with the test verifying that the ERROR log messages get printed - that can be seen in code review - but the test should verify that nothing gets deleted in misconfigured cases:

- missing deselector file
- empty deselector file
- deselector where clause is commented out

There is the case where someone writes a deselector that removes everything (eg `WHERE id IS NOT NULL` or `WHERE 'hello' = 'hello'`)... you can't feasibly defend against that and it might be the easiest correct way purge everything from a site anyway.